src/main/java/org/isf/admission/gui/AdmissionBrowser.java

```diff
@@ -1033,65 +1031,62 @@ private JPanel getWardPanel() {
 					}
 				}
 			}
-			wardBox.addActionListener(new ActionListener() {
-				public void actionPerformed(ActionEvent e) {
-					// set yProg
-					if (wardBox.getSelectedIndex() == 0) {
-						yProgTextField.setText("");
-						return;
+			wardBox.addActionListener(e -> {
+				// set yProg
+				if (wardBox.getSelectedIndex() == 0) {
+					yProgTextField.setText("");
+					return;
+				} else {
+					String wardId = ((Ward) wardBox.getSelectedItem()).getCode();
+					if (editing && wardId.equalsIgnoreCase(admission.getWard().getCode())) {
+						yProgTextField.setText("" + admission.getYProg());
 					} else {
-						String wardId = ((Ward) wardBox.getSelectedItem()).getCode();
-						if (editing && wardId.equalsIgnoreCase(admission.getWard().getCode())) {
-							yProgTextField.setText("" + admission.getYProg());
-						} else {
-							int nextProg = 1;
-							try {
-								nextProg = admissionManager.getNextYProg(wardId);
-							}catch(OHServiceException ex){
-                                OHServiceExceptionUtil.showMessages(ex);
-							}
-							yProgTextField.setText("" + nextProg);
-
-							// get default selected warn default beds number
-							int nBeds = ((Ward) wardBox.getSelectedItem()).getBeds();
-							int usedBeds = 0;
-							try {
-								usedBeds = admissionManager.getUsedWardBed(wardId);
-							}catch(OHServiceException ex){
-                                OHServiceExceptionUtil.showMessages(ex);
-							}
-							int freeBeds = nBeds - usedBeds;
-							if (freeBeds <= 0)
-								JOptionPane.showMessageDialog(AdmissionBrowser.this, MessageBundle.getMessage("angal.admission.wardwithnobedsavailable"));
+						int nextProg = 1;
+						try {
+							nextProg = admissionManager.getNextYProg(wardId);
+						} catch(OHServiceException ex){
+OHServiceExceptionUtil.showMessages(ex);
```

Comment:
what happens to the indentation here?

Reply:
that whole routine had very strange indentation to start; I'll reformat the lambda part of it.